### DISCUSSION TITLE
How to use this?

### DISCUSSION THREAD

**Discussion by kpoman** [RosettaDrone/rosettadrone#227] [General]
Hello,
Using a DJI mini 2.
I was able to build the apk and upload it to my phone (git cloned the latest repo).
I launch the app with RC connected. I do see live image from the cam inside the rosettadrone app.
I launch, on my PC, QGroundControl. After setting "use external device for GCS" to on, then adding my PC ip address to "Host 1", it seems like there is a connection, as I do see "Not ready" on QGroundControl, and it is receiving messages even though all of them seem lost.
What am I supposed to do? Is the apk supposed to establish, himself, connections to the PC? Which IP and ports should I set on the rosetta config page? There are host 1 to 4, video and WebRTC settings. Which should be set?
And why QGroundControl isnt able to receive the messages correctly?
I suppose there is a missing tutorial on how to set everything up.
Thank you

**Comment by kripper** [comment]
You have to set the telemetry port in RD (host 1 to 4) with the IP and the telemetry port of GCS:

> If you wish to use QGroundControl on an external device, click the Gear icon to access Settings, check Use GCS on an external device, then specify an IP address. Additionally a second telemetry connection con be configured to run AI scripts using DroneKit Python.

I'm afraid we don't have a WebRTC setup guide at the moment. Maybe @angel-ayala can provide you some hints.
Please send a pull request to enhance the documentation.

**Reply by kpoman** [reply]
Thanks for the reply! Each of those 4 settings controls something different from the drone? Where can I find out more info about this?
And about WebRTC, so, basically, I wont be able to see the live image out of the drone from the QGC PC app?

**Reply by kripper** [reply]
RD can send the telemetry data to multiple MAVLINK endpoints.
You will need to run a WebRTC viewer. I believe the tested viewer was in a different repo (not RD).

**Comment by angel-ayala** [comment]
Hi @kpoman ,

Yes indeed for the WebRTC you first need the server that will serve as a intermediate to present the WebRTC peers to do the handshake and ask for the streaming. You can use this repo as reference https://github.com/angel-ayala/webrtc-client-server/
I don't remember the exacts steps, but is something like as follows:

    Enable the WebRTC client in RD's settings and set the server IP address
    Run the WebRTC server and watch each peer ID.
    Execute the web client and fill the form with the RD WebRTC peer ID to ask for the stream.
    Receive the video feed in the client's browser.

Test it and let me know if you have any issues.

**Reply by kpoman** [reply]
@angel-ayala sorry if I misunderstood. Is RD generating live stream? It does need to send it to a peer on the network? That would be this WebRTC server? And then, from my PC, I would connect as client to that WebRTC server?

**Reply by angel-ayala** [reply]
> @angel-ayala sorry if I misunderstood. Is RD generating live stream?

Yes, RD can be a WebRTC peer that allows the streaming of the connected aircraft's camera feed.

> It does need to send it to a peer on the network? That would be this WebRTC server? And then, from my PC, I would connect as client to that WebRTC server?

I strongly recommend to read about WebRTC for better understanding, maybe this site could help https://www.techtarget.com/searchunifiedcommunications/definition/WebRTC-Web-Real-Time-Communications
